src/update_xs_yum.ml

```diff
@@ -435,29 +438,25 @@ let run root branch source_iso s3bin =
     (fun (binpkg_iso, sources_iso) ->
       Printf.printf "Running createtree\n%!";
       create_tree root binpkg_iso sources_iso
-      >>|= fun () ->
+      >>= fun () ->
       Printf.printf "Running createrepo\n%!";
-      check (Lwt_unix.system (Printf.sprintf "createrepo %s/domain0" root))
-      >>|= fun _ ->
+      check_command (Printf.sprintf "createrepo %s/domain0" root)
+      >>= fun () ->
       Printf.printf "Running s3cmd sync\n%!";
-      check (Lwt_unix.system (Printf.sprintf "s3cmd sync --delete-removed %s %s" root s3bin))
-      >>|= fun _ -> Lwt.return ok)
+      check_command (Printf.sprintf "s3cmd sync --delete-removed %s %s" root s3bin))
 
 let get_http_body url =
   Client.get (Uri.of_string url)
   >>= fun (res,body) ->
-  let ok = 
-    if not (res |> Response.status |> Code.code_of_status |> Code.is_success)
-    then
-      (
-        Printf.fprintf stderr "Bad response: %s\n" (res |> Response.status |> Code.string_of_status);
-        Lwt.return (`Error `Bad_response)
-      )
-    else Lwt.return (`Ok ())
-  in
-  ok
-  >>|= fun () ->
-  (Cohttp_lwt_body.to_string body >>= fun s -> Lwt.return (`Ok s))
+  if not (res |> Response.status |> Code.code_of_status |> Code.is_success)
+  then
+    (
+      let error = Printf.sprintf "Bad response: %s\n" (res |> Response.status |> Code.string_of_status) in
+      fail_with_log error
+    )
```

Comment:
would this be nicer with a `begin ... end` instead of `( ... )`?

Reply:
Sure, whichever people find more readable :smile:

Reply:
Done.